src/mo_def/syntax.ml

```diff
@@ -172,7 +172,7 @@ and exp' =
   | AssignE of exp * exp                       (* assignment *)
   | ArrayE of mut * exp list                   (* array *)
   | IdxE of exp * exp                          (* array indexing *)
-  | FuncE of string * sort_pat * typ_bind list * pat * typ option * sugar * exp  (* function *)
+  | FuncE of string * sort_pat * typ_bind list * pat * typ option * sugar * function_context * exp  (* function *)
```

Comment:
An alternative might have been to make the (mutable) function_context a custom annotation on the sort_pat (since we can update those already).

Just an idea, not saying it should be done.

Reply:
I find this a good idea. Thanks for raising